packages/gas-fee-controller/src/GasFeeController.test.ts

```diff
@@ -1001,6 +978,31 @@ describe('GasFeeController', () => {
       });
     });
     describe('when passed a networkClientId in options object', () => {
+      const getDefaultOptions = () => ({
+        getIsEIP1559Compatible: jest.fn().mockResolvedValue(true),
+        networkControllerState: {
```

Comment:
If we _do_ want to extract common setup to a function like this, then we should be careful that this matches the scenario we've describe and that all tests within the `describe` need this data to run. In this case, the `describe` says "when passed a `networkClientId` in `options` object". So the question is, `networksMetadata` need to be populated with all three of these networks in order to do that? Furthermore, does `getIsEIP1559Compatible` need to be `true`?

Reply:
Hi Elliot!

I understand from your comment that [the resetMocks PR](https://github.com/MetaMask/core/pull/4417) aims to make tests more self-documenting, which I fully support! Just want to make sure we’re keeping setup code concise and relevant to the tests. 

For example, as you mentioned, much of the configuration options are not relevant to the tests they are contained in. I appreciate you inlining the test setup variables. I think a more careful, case-by-case approach might give better results. I'm happy to discuss how that might be done and to assist with that effort.

In terms of the difficulties with merging and rebasing, as an example, I'm trying to revert [these changes to GasFeeController.test.ts](https://github.com/MetaMask/core/pull/4068/files#diff-46b4165a6fef6c662829473a7ae3dd5c12454ca7ea811b539c0e88dc464f7e3b) and it requires some care.

Reply:
Well the original goal of the `resetMocks` PR was simply to make sure that mock functions were being properly reset for each test. I may have been overzealous in making the changes I did to GasFeeController but I wanted to make sure that we didn't run into issues with mock functions again. Using a function instead of a variable to store common options as you've done in this PR also accomplishes that goal, so I am happy with that. If your changes also make it easier to merge future changes then it makes sense to do what you've done here as well. We can always review the tests on a case-by-case basis and remove unnecessary setup later.